zingo-sync/src/witness.rs

```diff
@@ -41,21 +43,76 @@ impl Default for ShardTrees {
 }
 
 /// Required data for updating [`shardtree::ShardTree`]
-pub struct ShardTreeData {
+pub(crate) struct WitnessData {
     pub(crate) sapling_initial_position: Position,
     pub(crate) orchard_initial_position: Position,
-    pub(crate) sapling_leaves_and_retentions: Vec<(Node, Retention<BlockHeight>)>,
+    pub(crate) sapling_leaves_and_retentions: Vec<(sapling_crypto::Node, Retention<BlockHeight>)>,
     pub(crate) orchard_leaves_and_retentions: Vec<(MerkleHashOrchard, Retention<BlockHeight>)>,
 }
 
-impl ShardTreeData {
+impl WitnessData {
     /// Creates new ShardTreeData
     pub fn new(sapling_initial_position: Position, orchard_initial_position: Position) -> Self {
-        ShardTreeData {
+        WitnessData {
             sapling_initial_position,
             orchard_initial_position,
             sapling_leaves_and_retentions: Vec::new(),
             orchard_leaves_and_retentions: Vec::new(),
         }
     }
 }
+
+/// Located prunable tree data built from nodes and retentions during scanning for insertion into the shard store.
+pub struct LocatedTreeData<H> {
```

Comment:
Why use this new type? BatchInsertionResult contains these two fields, plus extras that may-or-may-not be useful later...but adding this transformation that only strips data from the result before returning it feels like unneeded ceremony.

Reply:
I actually tried this approach but I had issues with the fact that BatchInsertionResult's remainder field is an Iterator. I spent quite a bit of time trying to get it to work but settled for creating a new struct with all the currently relevant data.